### ISSUE TITLE
Permissions issue on new file upload

### ISSUE THREAD

**Issue by FurmanSK** [Equals182/eqFTP#272]
## Steps to Reproduce (for bugs)
1.

## Debug Information

* eqFTP version: Latest beta
* Brackets version: 1.8
* Operating System and version:

On upload of a new file, the files permissions are set to 0640 which before we used to be able to set the permissions of a file on upload right? Being able to set permissions to say 655 or whatever we need is important. Otherwise I have to putty into my server and chmod the new file to what it needs to be.

Thanks, keep up the good work!

-Furman

**Comment by Equals182** [comment]
No, there was no option to change file's permissions in any versions. In fact, I did a tiny research now and I don't even know how to implement it yet. What I found though is that you *may* have so called "umask" for user on sftp server (if you use sftp of course). 
Did you have same issue with previous versions of eqftp?

**Comment by FurmanSK** [comment]
Oh thought it had permissions setting. Might be thinking of something else then haha. 

I was using FTP but just changed to SFTP. Would that make a difference?

**Comment by Equals182** [comment]
I'd like to add permission settings in future, I just need to figure out how to do it 😄 
As for SFTP over FTP, it *may* make difference, but I can't tell for sure, unfortunately, you gotta check it out :)

**Comment by FurmanSK** [comment]
So looked at umask and it was set to default 0002. Changed it in my .profile for my linux server and using SFTP now it will set files to 0644. So this probably shouldn't be a high priority task. But might be cool for a future release down the road. Haha.

Thanks again.

-Furman

**Comment by Equals182** [comment]
So, as for permission settings, I have a [card for it](https://github.com/Equals182/eqFTP/projects/1#card-1298116).
And for permissions on upload, I'm not quite sure if it's even possible, but I'll try to find a way, maybe I'll set permissions after uploading, not sure yet.

**Comment by pipiscrew** [comment]
@Equals182 **chmod** check @ [jsftp](https://github.com/sergi/jsftp/issues/101)

**Comment by Equals182** [comment]
yeah with jsftp you can send any raw command, I know. But what about ssh2? I didn't see there any raw commands or special chmod command.

**Comment by pipiscrew** [comment]
in vquick search I _maybe_ found that using the function **.once()** to send the commands. you have to extend it as per wish. But I dont think that the SSH is a priority.

https://github.com/wdavidw/node-ssh2-fs/search?utf8=%E2%9C%93&q=chmod

keep up

**Comment by drzraf** [comment]
* sftp usually allow the run the chmod/umask command in other to change perms of remote files.
* sftp-server allow forcing a umask (server-side) which is a set of MAXIMUM permissions.
* (there no such thing as "minimal permissions" server-side)
